src/decoder/mod.rs

```diff
@@ -347,12 +347,12 @@ pub struct Reader<R: Read> {
     /// control chunk. The IDAT image _may_ have such a chunk applying to it.
     fctl_read: u32,
     next_frame: SubframeIdx,
-    /// Previous raw line
-    prev: Vec<u8>,
-    /// Current raw line
-    current: Vec<u8>,
-    /// Start index of the current scan line.
-    scan_start: usize,
+    /// Vec containing the uncompressed image data currently being processed.
+    data_stream: Vec<u8>,
+    /// Index in `data_stream` where the previous row starts.
+    prev_start: usize,
+    /// Index in `data_stream` where the current row starts.
```

Comment:
I wonder if it would be useful to point out that

* 1) `prev_start` points at pixel data - `rowlen - 1` worth of bytes, but 2) `current_start` points filter kind byte + pixel data (potentially incomplete - at most `rowlen` worth of bytes).
* `prev_start == current_start` indicates that there is no previous row